src/Core/Types.Filters/FilterInputTypeDescriptor.cs

```diff
@@ -183,6 +191,22 @@ public class FilterInputTypeDescriptor<T>
                 nameof(propertyOrMethod));
         }
 
+        public IObjectFilterFieldDescriptor<TObject> Filter<TObject>(
+            Expression<Func<T, TObject>> propertyOrMethod) where TObject : class
```

Comment:
I wonder if there is a better generic constraint than this...

Reply:
And also what about interfaces or structs that represent a complex type.

Reply:
Maybe the name is wrong.... is it really a filter? I mean in the case of an object?

Reply:
And what if an object is comparable :D